crypto/evp/evp_fetch.c

```diff
@@ -259,7 +259,17 @@ inner_evp_generic_fetch(struct evp_method_data_st *methdata,
 {
     OSSL_METHOD_STORE *store = get_evp_method_store(methdata->libctx);
     OSSL_NAMEMAP *namemap = ossl_namemap_stored(methdata->libctx);
+#ifdef FIPS_MODULE
+    /*
+     * The FIPS provider has its own internal library context where only it
+     * is loaded.  Consequently, property queries aren't relevant because
+     * there is only one fetchable algorithm and it is assumed that the
+     * FIPS-ness is handled by the using algorithm.
+     */
```

Comment:
there are a few unapproved algorithms in fips mode.. I  cant see this being a problem though??

Reply:
The only fetches from the internal lib ctx come from the FIPS provider itself, so it should be fine.
If an algorithm needs a something non-approved, it's up to the algorithm to handle the consequences.  In this case, the ECX curves are not approved but are allowed to be used as part of the hybrid so it's okay.